example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,188 @@
+/* Copyright 2023 Benjamin Worpitz, Erik Zenker, Bernhard Manfred Gruber, Jan Stephan
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+
+//! Convolution Example
+//!
+//! 1D convolution example: Creates two 1D arrays, applies convolution filter.
+//! Array sizes are hardcoded.
+//!
+
+/**
+ * @brief The ConvolutionKernel function-object
+ * Calculates 1D convolution using input and filter arrays.
+ */
+struct ConvolutionKernel
+{
+    /** @brief Main convolution code
+     *  @param Accelerator
+     *  @param Input array, first input of convolution integral
+     *  @param Filter array, second input of convolution integral
+     *  @param Empty output array to be filled
+     *  @param Input array size
+     *  @param Filter size
+     */
+    template<typename TAcc, typename TElem>
+    ALPAKA_FN_ACC auto operator()(
+        TAcc const& acc,
+        TElem const* const input,
+        TElem const* const filter,
+        TElem* const output,
+        const std::size_t inputSize,
+        const std::size_t filterSize) const -> void
+    {
+        using Idx = alpaka::Idx<TAcc>;
+        using Vec = alpaka::Vec<alpaka::Dim<TAcc>, Idx>;
+        using VecInt = alpaka::Vec<alpaka::DimInt<1u>, Idx>;
+
+
+        auto const globalThreadIdxX = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0];
+        Vec const globalThreadExtent = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc);
+
+        // Since the kernel is launched 1-D calculating linearizedGlobalThreadIdx line is unnecessary.
+        // globalThreadIdx[0] can be used to map all the threads.
+        if(globalThreadIdxX < inputSize)
+        {
+            int32_t const halfFilterSize = filterSize / 2;
+            TElem result = 0.0f;
+            // Calculate sum of multiplications of corresponding elements
+            auto const start
+                = static_cast<int32_t>(std::max(static_cast<int32_t>(globalThreadIdxX) - halfFilterSize, 0));
+            auto const stop = std::min(globalThreadIdxX + halfFilterSize, inputSize - 1);
+            for(int32_t i = start; i <= stop; ++i)
+                result += input[i] * filter[i + halfFilterSize - static_cast<int32_t>(globalThreadIdxX)];
+            output[globalThreadIdxX] = result;
+        }
+    }
+};
+
+/**
+ * @brief Function template to compare floating point values
+ *
+ */
+auto FuzzyEqual(float a, float b) -> bool
```

Comment:
```suggestion
auto FuzzyEqual(float a, float b) -> bool
```
It's no longer a `template` and I would also not need the comment, since the function name suggests what the function is about.